system/timedatectl.py

```diff
@@ -0,0 +1,173 @@
+  sample:
+    [
+      {
+        "name"     : "Timezone",
+        "old_value": "UTC (UTC, +0000)",
+        "new_value": "Asia/Tokyo (JST, +0900)",
+        "command"  : "timedatectl set-timezone Asia/Tokyo",
+        "changed"  : True
+      },
+      {
+        "name"     : "NTP enabled",
+        "old_value": "yes",
+        "new_value": "yes",
+        "command"  : "",
+        "changed"  : False
+      }
+    ]
+'''
+
+EXAMPLES = '''
+- name: set timezone to Asia/Tokyo
+  timedatectl: timezone=Asia/Tokyo
+'''
+
+
+def main():
+    # Arguments setting
+    # * 'name' is not required key to construct AnsibleModule,
+    #   but it is used to check if change will happen.
+    arguments = dict(
+        ntp      = dict(default=None, required=False, type='bool', name='NTP enabled'),
+        time     = dict(default=None, required=False, type='str',  name='Local time'),
+        timezone = dict(default=None, required=False, type='str',  name='Timezone'),
+    )
+    changes = []
+
+    # Construct 'module'
+    module = AnsibleModule(
+        argument_spec       = arguments,
+        supports_check_mode = True,
+        required_one_of     = [ arguments.keys() ],
+    )
+
+    # Get the current state from the stdout of status the command
+    old_state = module.run_command('timedatectl status', check_rc=True)[1]
+    # module.exit_json(changed=False, msg=old_state) # For debug
+
+    changed_any = False
+    for arg, spec in arguments.iteritems():
+        if module.params[arg] is not None:
+            if spec['type'] != 'bool':
+                new_value = module.params[arg]
+            else:
+                # new_value = 'yes' if module.params[arg] else 'no'
+                if module.params[arg]:
+                    new_value = 'yes'
+                else:
+                    new_value = 'no'
+            # Check the old value, which is written in `old_state`.
+            # Below regex is to find the value.
+            regex = re.compile(spec['name']+r':\s*([^\n]+)\n')
+            old_value = regex.search(old_state).group(1)
+            if new_value not in old_value:
+                # then, changes should happen!
+                command = 'timedatectl set-%s %s' % (arg, new_value)
+                changed = True
+                changed_any = True
+            else:
+                command = ''
+                changed = False
+            changes.append(
+                dict(
+                    name      = spec['name'],
+                    new_value = new_value,
+                    old_value = old_value,
+                    command   = command,
+                    changed   = changed
+                )
+            )
+
+    # Make changes
+    if not module.check_mode:
+        for change in changes:
+            if change['changed']:
+                module.run_command(change['command'], check_rc=True)
+        # Update 'new_value' field
+        new_state = module.run_command('timedatectl status', check_rc=True)[1]
```

Comment:
we want to use find_bin function instead of relying that it is in the PATH as that might change due to how ansible logs in.